File: C#/src/Implementations/SaveConfig.cs
using System;
using System.IO;

public class SaveConfig
{
    public static string GetSaveDirectory()
    {
        return "data/saves";
    }

    public static string GetAutoSaveDirectory()
    {
        return "data/saves/autosaves";
    }

    public static bool CreateDirectoryIfNotExists(string path)
    {
        try
        {
            if (!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
                return true;
            }
            return true;
        }
        catch (Exception e)
        {
            Console.WriteLine("Blad tworzenia katalogu " + path + ": " + e.Message);
            return false;
        }
    }
}

File: C++/src/Implementations/SaveConfig.cpp
#include "SaveConfig.h"
#include <filesystem>
#include <iostream>
#include <string>

std::string SaveConfig::GetSaveDirectory() { return "data/saves"; }
std::string SaveConfig::GetAutoSaveDirectory() { return "data/saves/autosaves"; }
bool SaveConfig::CreateDirectoryIfNotExists(const std::string& path)
{
    try
    {
        if (!std::filesystem::exists(path)) { return std::filesystem::create_directories(path); }
        return true;
    }
    catch (const std::filesystem::filesystem_error& e)
    {
        std::cout << "Blad tworzenia katalogu " << path << ": " << e.what() << '\n';
        return false;
    }
}